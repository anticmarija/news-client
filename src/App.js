import React, { useEffect } from "react";
import "./App.css";

import { Route, Switch, Redirect, useLocation } from "react-router-dom";

import Menu from "./components/menu/Menu";
import TopNews from "./components/top-news/TopNews";
import Categories from "./components/categories/Categories";
import Search from "./components/search/Search";
import Article from "./components/article/Article";
import config from "./config";
import { useDispatch, useSelector } from "react-redux";
import {
  setCountries,
  setActiveCountry,
} from "./store/actions/countries.actions";
import { setCategories } from "./store/actions/categories.actions";
import Category from "./components/category/Category";
import { findCountryIdInPathname } from "./utils/helpers";

function App() {
  const dispatch = useDispatch();
  const { activeCountry } = useSelector((state) => state.countries);

  const { pathname } = useLocation();

  useEffect(() => {
    const countries = config.getCountries();
    const categories = config.getCategories();

    dispatch(setCountries(countries));
    dispatch(setCategories(categories));

    const countryInRoute = findCountryIdInPathname(pathname);

    if (countryInRoute === -1) {
      //handle landing on page - set first country as default
      dispatch(setActiveCountry(countries[0].countryId));
    } else {
      //handle refresh with country in route
      dispatch(setActiveCountry(countryInRoute));
    }
    // eslint-disable-next-line
  }, []);

  return (
    activeCountry && (
      <div className="App">
        <Menu />
        <Switch>
          <Route exact path="/">
            <Redirect to={`/country/${activeCountry}`} />
          </Route>
          <Route path={`/country/:countryId/categories/:categoryName`}>
            <Category />
          </Route>
          <Route path={`/country/:countryId/categories`}>
            <Categories />
          </Route>
          <Route path={`/country/:countryId/search`}>
            <Search />
          </Route>
          <Route path={`/country/:countryId/articles/:articleId`}>
            <Article />
          </Route>
          <Route path="/country/:countryId">
            <TopNews />
          </Route>
        </Switch>
      </div>
    )
  );
}

export default App;
