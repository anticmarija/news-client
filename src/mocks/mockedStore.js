import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
export const initialState = {
  topNews: {},
  countries: {
    supportedCountries: [{ countryId: "gb", countryName: "Gb" }],
  },
  categories: {},
  article: {
    activeArticle: {
      title: "active",
      urlToImage:
        "https://i.dailymail.co.uk/1s/2020/06/17/16/29719992-0-image-a-23_1592408042536.jpg",
      description: "desc",
    },
  },
};

export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
