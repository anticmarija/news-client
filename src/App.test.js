import React from "react";
import App from "./App";
import { mockedStore } from "./mocks/mockedStore";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (s) => s }),
}));

describe("Test App component", () => {
  it("should render correctly", () => {
    const store = mockedStore();
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
