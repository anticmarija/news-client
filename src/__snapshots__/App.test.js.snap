// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component should render correctly 1`] = `
<div>
  <div
    class="Menustyle__MenuWrapperStyle-sc-1neq1ee-0 kTCkyP"
  >
    <div
      class="Menustyle__MenuItemsWrapperStyle-sc-1neq1ee-1 eEekmm"
    >
      <div
        class="Menustyle__MenuItemStyle-sc-1neq1ee-2 cVYhuA"
      >
        <a
          aria-current="page"
          class="Menustyle__LinkStyle-sc-1neq1ee-3 ccUwVd active"
          href="/country/gb/"
        >
          Top news
        </a>
      </div>
      <div
        class="Menustyle__MenuItemStyle-sc-1neq1ee-2 iJVOUw"
      >
        <a
          class="Menustyle__LinkStyle-sc-1neq1ee-3 ccUwVd"
          href="/country/gb/categories"
        >
          Categories
        </a>
      </div>
      <div
        class="Menustyle__MenuItemStyle-sc-1neq1ee-2 iJVOUw"
      >
        <a
          class="Menustyle__LinkStyle-sc-1neq1ee-3 ccUwVd"
          href="/country/gb/search"
        >
          Search
        </a>
      </div>
    </div>
    <div
      class="Menustyle__MenuCountriesWrapperStyle-sc-1neq1ee-4 hvGaFl"
    >
      <div
        class="Menustyle__MenuItemStyle-sc-1neq1ee-2 cVYhuA"
      >
        <a
          active="active"
          aria-current="page"
          class="Menustyle__LinkStyle-sc-1neq1ee-3 ccUwVd active"
          data-testid="country-link"
          href="/country/gb/"
        >
          GB 
        </a>
      </div>
    </div>
  </div>
  <div
    class="Layoutstyle__LayoutStyle-fqkpn0-0 dWQSdv"
  >
    <div>
      <h1
        class="Sharedstyle__MainHeaderStyle-sc-1a3bnp9-0 chhIjm"
      >
        Top news from 
        Gb
      </h1>
      <div
        class="Sharedstyle__NewsWrapperStyle-sc-1a3bnp9-1 jTferJ"
      >
        <div
          class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
        >
          <h2
            class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
          >
            article
          </h2>
          <div
            class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 eTaMhC"
          />
          <div
            class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 dPDhAr"
          >
            <div>
              desc
            </div>
            <span
              class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
            >
              <a
                data-testid="more-link"
                href="/country/gb/articles/article"
              >
                More &gt;
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
